src/layer/vector/Canvas.js

```diff
@@ -339,7 +339,8 @@ export var Canvas = Renderer.extend({
 
 		if (options.stroke && options.weight !== 0) {
 			if (ctx.setLineDash) {
-				ctx.setLineDash(layer.options && layer.options._dashArray || []);
+				ctx.lineDashOffset = Number(options.dashOffset || 0);
```

Comment:
Is `Number` really neaded? We don't normaly use this in Leaflet

```suggestion
				ctx.lineDashOffset = options.dashOffset || 0;
```

Reply:
a) Leaflet docs say dashOffset is string; lineDashOffset should be float
b)  _dashArray is processed the same way. see [Canvas._updateDashArray](https://github.com/Leaflet/Leaflet/blob/e9fc60c022fab6557744576f324c82ae1958b16c/src/layer/vector/Canvas.js#L195)

Reply:
Interessting ... you are right 👍